### DISCUSSION TITLE
Meaning of `expression = expression_sum = expression = df[str(self.expression)]` ?

### DISCUSSION THREAD

**Discussion by yohplala** [vaexio/vaex#1851] [Q&A]
Hi @maartenbreddels 
I am sorry, this is a pure python question.
I could find this line of code in `AggregatorDescriptorMean.add_tasks()`, [line 355](https://github.com/vaexio/vaex/blob/ee09761251eb93574edcc5f78b0ad8f872228137/packages/vaex-core/vaex/agg.py#L355).
`expression = expression_sum = expression = df[str(self.expression)]`

Googling chained assignments in Python, I could find [this wikipedia ressource](https://en.wikipedia.org/wiki/Assignment_%28computer_science%29#Chained_assignment) explaining:
_The assignments are executed left-to-right so that `i = arr[i] = f()` evaluates the expression `f()`, then assigns the result to the leftmost target `i`, and then assigns the same result to the next target `arr[i]`, using the new value of `i`. This is essentially equivalent to `tmp = f(); i = tmp; arr[i] = tmp`_

Applied to this row in `add_task()`, it would translate if I am right into:
```python
expression = df[str(self.expression)]
expression_sum = expression
expression = expression
```
I don't understand why assigning twice the same value to `expression`. Please, what am I missing?
Thanks for your reply.
Have a good day,
Bests,

**Comment by maartenbreddels** [comment]
Ah, I've explained this a bit #1848, and it's indeed code, it doesn't do much, except the full rightmost expression is evaluated, and indeed assigned multiple times. If you want to open a PR to clean up that code that would be appreciated.

**Reply by yohplala** [reply]
> Ah, I've explained this a bit #1848, and it's indeed code, it doesn't do much, except the full rightmost expression is evaluated, and indeed assigned multiple times. If you want to open a PR to clean up that code that would be appreciated.

Sorry, I am just seeing your answer in #1834 . Ok, crystal clear. Yes I can clear this of course. Will not do it right now, but will do it this week-end (sorry, I am not on my dev computer at  the moment)
Thanks a lot for your feedback!
Bests,
